segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/WriteOperation.java

```diff
@@ -54,35 +54,41 @@ public void run() throws BadOffsetException, IOException, StorageNotPrimaryExcep
         FileDescriptor lastFile = handle.getLastFile();
 
         Timer timer = new Timer();
-        try (FSDataOutputStream stream = this.context.fileSystem.append(lastFile.getPath())) {
-            if (this.offset != lastFile.getLastOffset()) {
-                // Do the handle offset validation here, after we open the file. We want to throw FileNotFoundException
-                // before we throw BadOffsetException.
-                throw new BadOffsetException(handle.getSegmentName(), lastFile.getLastOffset(), this.offset);
-            } else if (stream.getPos() != lastFile.getLength()) {
-                // Looks like the filesystem changed from underneath us. This could be our bug, but it could be something else.
-                // Update our knowledge of the filesystem and throw a BadOffsetException - this should cause upstream code
-                // to try to reconcile; if it can't then the upstream code should shut down or take other appropriate measures.
-                log.warn("File changed detected for '{}'. Expected length = {}, actual length = {}.", lastFile, lastFile.getLength(), stream.getPos());
-                lastFile.setLength(stream.getPos());
-                throw new BadOffsetException(handle.getSegmentName(), lastFile.getLastOffset(), this.offset);
-            }
+        try {
```

Comment:
This method is getting long and deep. Perhaps a private helper method is in order.